src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/Formatting/FormattingHandler.ts

```diff
@@ -0,0 +1,63 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { RequestType } from 'vscode-languageclient';
+import { RazorDocumentManager } from '../RazorDocumentManager';
+import { RazorLanguageServerClient } from '../RazorLanguageServerClient';
+import { RazorLogger } from '../RazorLogger';
+import { convertTextEditToSerializable, SerializableTextEdit } from '../RPC/SerializableTextEdit';
+import { SerializableFormattingParams } from './SerializableFormattingParams';
+import { SerializableFormattingResponse } from './SerializableFormattingResponse';
+
+export class FormattingHandler {
+    private static readonly provideFormattingEndpoint = 'textDocument/formatting';
+    private formattingRequestType: RequestType<SerializableFormattingParams, SerializableFormattingResponse, any> = new RequestType(FormattingHandler.provideFormattingEndpoint);
+    private emptyFormattingResponse = new SerializableFormattingResponse();
+
+    constructor(
+        private readonly documentManager: RazorDocumentManager,
+        private readonly serverClient: RazorLanguageServerClient,
+        private readonly logger: RazorLogger) {
+    }
+
+    public register() {
+        // tslint:disable-next-line: no-floating-promises
+        this.serverClient.onRequestWithParams<SerializableFormattingParams, SerializableFormattingResponse, any>(
```

Comment:
I know this is the pattern we follow everywhere, but do we have an issue to maybe get rid of the floating promises in the first place?

Reply:
Created an issue here: https://github.com/dotnet/razor-tooling/issues/6908